R/utilities.R

```diff
@@ -282,3 +278,70 @@ expand_patients <- function(patients, all_pts) {
     )
     return(patients)
 }
+
+
+
+#' Decompose Patients into Relevant Components
+#'
+#' This function takes in a character vector or list of patients and decomposes it into a
+#' structured format.
+#'
+#' The primary use of this function is to correctly setup indexing variables for
+#' predicting survival quantities (see [`predict(SurvivalSamples)`][SurvivalSamples-class])
+#'
+#' @param patients (`character` or `list`) patient identifiers. If `NULL` will be set to `all_pts`.
+#'
+#' @param all_pts (`character`) the set of allowable patient identifiers.
+#' Will cause an error if any value of `patients` is not in this vector.
+#'
+#' @return A list containing three components:
+#' \itemize{
+#'   \item \code{patients_list}: (`list`) each element is named and is a character vector
```

Comment:
```suggestion
#'   \item `patients_list`: (`list`) each element is named and is a character vector
```
should work? I don't think we should use `\code{}` et al because we use the md syntax

Reply:
re: names - maybe just omit the "patients_" prefix for all of these?